pyfuncol/dict.py

```diff
@@ -54,7 +78,8 @@ def flat_map(self: Dict[A, B], f: Callable[[Tuple[A, B]], Dict[C, D]]) -> Dict[C
     Returns:
         The new dict.
     """
-    res = {}
+    res = deepcopy(self)
+    res.clear()
```

Comment:
Couldn't we just cast it at the end?

Reply:
The issue with that is that subclasses that have a constructor that takes parameters cannot be used (like defaultdict). So if we instead copy the receiver we are sure the returning object is already initialized. It makes however the code ugly and less performant, so maybe we simply support simple subclasses (like OrderedDict) and don't care about the rest?

Reply:
oh I see... Yes, I am a bit scared about performance when called on large data, like several MB to GB list or dict, this will take a while and a lot of memory for the deep copy to finish, then to be cleared.
I think that simple subclasses for now are sufficient, until we find a better solution. I do not think the performance tradeoff is worth it for now. WDYT?

Reply:
Yes, let's support only the simple ones for now, I'll do the changes and merge master on the branch